codec/src/main/java/io/netty/handler/codec/base64/Base64.java

```diff
@@ -168,31 +173,59 @@ private static void encode3to4(
         // significant bytes passed in the array.
         // We have to shift left 24 in order to flush out the 1's that appear
         // when Java treats a value as negative that is cast from a byte to an int.
-        int inBuff =
-                (numSigBytes > 0? src.getByte(srcOffset)     << 24 >>>  8 : 0) |
-                (numSigBytes > 1? src.getByte(srcOffset + 1) << 24 >>> 16 : 0) |
-                (numSigBytes > 2? src.getByte(srcOffset + 2) << 24 >>> 24 : 0);
+
+        // Packing bytes into an int to reduce bound and reference count checking.
+        final int value;
+        switch (numSigBytes) {
+            case 3:
+                value = (alphabet[inBuff >>> 18       ] << 24) |
+                        (alphabet[inBuff >>> 12 & 0x3f] << 16) |
+                        (alphabet[inBuff >>>  6 & 0x3f] << 8)  |
+                        alphabet[inBuff        & 0x3f];
+                break;
+            case 2:
+                value = (alphabet[inBuff >>> 18       ] << 24) |
+                        (alphabet[inBuff >>> 12 & 0x3f] << 16) |
+                        (alphabet[inBuff >>> 6  & 0x3f] << 8)  |
+                        EQUALS_SIGN;
+                break;
+            case 1:
+                value = (alphabet[inBuff >>> 18       ] << 24) |
+                        (alphabet[inBuff >>> 12 & 0x3f] << 16) |
+                        (EQUALS_SIGN << 8)                     |
+                        EQUALS_SIGN;
+                break;
+            default:
+                // NOOP
+                return;
+        }
+        if (dest.order() == ByteOrder.BIG_ENDIAN) {
+            dest.setInt(destOffset, value);
+        } else {
+            dest.setIntLE(destOffset, value);
```

Comment:
I would prefer avoiding the double swap but I agree it isn't necessary for this PR (fine for followup PR, or otherwise delay). Since we are already doing pretty low level byte manipulation here and it has been shown to have a noticeable performance gain I think handling the endianness in `Base64` is reasonable.